runtime/src/weights/pallet_creditcoin.rs

```diff
@@ -0,0 +1,183 @@
+
+//! Autogenerated weights for `pallet_creditcoin`
+//!
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
+//! DATE: 2022-04-05, STEPS: `50`, REPEAT: 30, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! EXECUTION: Some(Wasm), WASM-EXECUTION: Compiled, CHAIN: Some("dev"), DB CACHE: 1024
+
+// Executed Command:
+// ./target/release/creditcoin-node
+// benchmark
+// --chain
+// dev
+// --steps=50
+// --repeat=30
+// --pallet
+// pallet_creditcoin
+// --extrinsic=*
+// --execution
+// wasm
+// --wasm-execution=compiled
+// --heap-pages=10000
+// --output
+// ./runtime/src/weights/pallet_creditcoin.rs
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::{traits::Get, weights::Weight};
+use sp_std::marker::PhantomData;
+
+/// Weight functions for `pallet_creditcoin`.
+pub struct WeightInfo<T>(PhantomData<T>);
+impl<T: frame_system::Config> pallet_creditcoin::WeightInfo for WeightInfo<T> {
+	// Storage: Creditcoin DealOrders (r:65 w:64)
+	// Storage: Creditcoin BidOrders (r:0 w:255)
+	// Storage: Creditcoin AskOrders (r:0 w:128)
+	// Storage: Creditcoin Offers (r:0 w:128)
+	// Storage: Creditcoin UnverifiedTransfers (r:0 w:1)
+	fn on_initialize(a: u32, b: u32, o: u32, d: u32, f: u32, u: u32, ) -> Weight {
+		(0 as Weight)
+			// Standard Error: 124_000
+			.saturating_add((4_232_000 as Weight).saturating_mul(a as Weight))
+			// Standard Error: 611_000
+			.saturating_add((16_589_000 as Weight).saturating_mul(d as Weight))
+			// Standard Error: 1_575_000
+			.saturating_add((17_031_000 as Weight).saturating_mul(f as Weight))
+			// Standard Error: 1_575_000
+			.saturating_add((11_551_000 as Weight).saturating_mul(u as Weight))
+			.saturating_add(T::DbWeight::get().reads(1 as Weight))
+			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(d as Weight)))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(a as Weight)))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(b as Weight)))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(o as Weight)))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(d as Weight)))
+	}
+	// Storage: Creditcoin Addresses (r:1 w:1)
+	fn register_address(b: u32, e: u32, ) -> Weight {
+		(18_824_000 as Weight)
+			// Standard Error: 0
+			.saturating_add((6_000 as Weight).saturating_mul(b as Weight))
+			// Standard Error: 0
+			.saturating_add((5_000 as Weight).saturating_mul(e as Weight))
+			.saturating_add(T::DbWeight::get().reads(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+	}
```

Comment:
If I understand this correctly you are taking an initial weight here and then augment it with the number of reads and writes for each extrinsic. Is that correct ? 

I am worried that this is disconnected with the extrinsic functions and whenever we adjust behavior for some of the extrinsics we need not forget to update the weight formula too. This process is very prone to errors. Ideally there will be a way to connect both extrinsic behavior implementation and extrinsic weight implementation. 

Note: this issue was present in the old implementation too but was less obvious b/c the macros defining the weights were directly above the function definitions.

Reply:
This code is automatically generated by the benchmarks, so we just need to re-benchmark whenever we change behavior to update these weights

Reply:
As Nathan said. After every extrinsic change we need to rerun the script I provided with options -cb, c to recompile and -b to benchmark. It will auto generate the updated files with the new weights. We need to push that commit with every major extrinsic changes.

Reply:
FTR it's this command: https://github.com/gluwa/creditcoin/pull/126/files#diff-4818d880e3f3808d9acaaff75c745bbba68923186bae4b93e5e6cdfd443dc6f9R8

Reply:
> How does this happen exactly ? e.g. what command? We can add this to the CI config and do a `git diff` afterwards. If nothing changed then we're good.

Benchmarks will always output different data, is this a limitation? Even after pushing updated weights CI will probably complain.